src/Mix.php

```diff
@@ -74,13 +83,46 @@ public function __invoke($path, $manifestDirectory = ''): string
         }
 
         $manifest = self::$manifests[$manifestPath];
-        if (! isset($manifest[$path])) {
+
+        $path = self::resolvePath($path);
+
+        if (!isset($manifest[$path])) {
             throw new Exception("Unable to locate AssetMix file: {$path}.");
         }
 
         return $urlDomain . $manifestDirectory . $manifest[$path];
     }
 
+    /**
+     * This will strip the subdir part off of /subdir/js/app.js?1682657485
+     * 
+     * If asset timestamping is enabled then it will remove the timestamp 
+     * query string
+     * 
+     * '/subdir/js/app.js?1682657485'
+     * 
+     * will become
+     * '/js/app.js' 
+     * 
+     * @param mixed $path 
+     * @return mixed 
+     */
+    private static function resolvePath($path)
+    {
+        $pattern = sprintf(
+            '@^%s@',
+            self::$request->getAttribute('base')
```

Comment:
Alternate approach here would be to get base value via `Configure::read('App.base')`. The benefit of having this is that we can remove the `$request` dependency, so constructor required.

Reply:
Thank you. A way better approach